adapter-kafka/src/main/java/io/nosqlbench/adapter/kafka/ops/OpTimeTrackKafkaConsumer.java

```diff
@@ -157,15 +167,30 @@ public void onComplete(Map<TopicPartition, OffsetAndMetadata> map, Exception e)
                             }
 
                             resetManualCommitTrackingCnt();
-                        } else {
+                        } else  {
+                            updateE2ELatencyMetric(record);
                             incManualCommitTrackingCnt();
                         }
                     }
+                    updateE2ELatencyMetric(record);
                 }
             }
         }
     }
 
+    private void updateE2ELatencyMetric(ConsumerRecord<String, String> record) {
+        long startTimeStamp = 0L;
+        switch (e2eStartingTimeSrc) {
+            case MESSAGE_PUBLISH_TIME:
+                startTimeStamp = record.timestamp();
```

Comment:
how is r`ecord.timestamp` populated ?

Reply:
From the Kafka API Doc. The NB Kafka producer uses the default setting and doesn't explicitly set a timestamp.

```
The record also has an associated timestamp. If the user did not provide a timestamp, the producer will stamp the record with its current time. The timestamp eventually used by Kafka depends on the timestamp type configured for the topic.

If the topic is configured to use CreateTime, the timestamp in the producer record will be used by the broker. 
```

Reply:
Here is the Kafka API Doc, referenced by Yabin: https://kafka.apache.org/34/javadoc/org/apache/kafka/clients/producer/ProducerRecord.html.  I will make sure to configure the topic with `CreateTime`.

Reply:
It seems this can be done by adding `message.timestamp.type: CreateTime` to the kafka_config.properties. But, to my understanding, that file is not used. At least nothing seems to be using the result stored in this variable https://github.com/nosqlbench/nosqlbench/blob/e832e5a5897978d1b46e01bc5e17b29e6f61f5ba/adapter-kafka/src/main/java/io/nosqlbench/adapter/kafka/dispensers/KafkaBaseOpDispenser.java#L79. 

Anyway according to [the doc](https://docs.confluent.io/platform/current/installation/configuration/topic-configs.html#message-timestamp-type) the default is `CreateTime`. So, we should be good.